osu.Server.Spectator/Hubs/MultiplayerHubContext.cs

```diff
@@ -128,26 +128,27 @@ public async Task ChangeAndBroadcastUserState(ServerMultiplayerRoom room, Multip
         {
             log(room, user, $"User state changed from {user.State} to {state}");
 
-            user.State = state;
-
             string? connectionId = users.GetConnectionIdForUser(user.UserID);
 
             if (connectionId != null)
             {
-                switch (state)
+                if (state == MultiplayerUserState.FinishedPlay)
```

Comment:
Please change this back to a switch statement.

Reply:
Does it make sense to have a switch with only 1 case? There needs to be one check for previous state and one for new state.

Reply:
I believe this is incorrect. I don't understand what you mean. The progression between states is sequential.